scripts/travis/update/pr-creator.js

```diff
@@ -0,0 +1,134 @@
+        this.github = new GitHub({token});
+        this.repoOrigin = this.github.getRepo(githubUser, ORIGIN_REPO);
+        this.repoDestination = this.github.getRepo(githubUser, githubRepo);
+        this.issue = this.github.getIssues(githubUser, githubRepo);
+        this.commit = commit;
+    }
+
+    async createOrUpdate(title, head, base, commit) {
+        const { data: prs } = await this.repoDestination.listPullRequests({ state: 'open', head: `${ORGANISATION}:${head}`, base });
+
+        if (prs.length < 1) {
+            const { data: pr } = await this.repoDestination.createPullRequest({ title, head, base, body: `sha:${commit}` });
+            return pr.number;
+        } else {
+            const upstreamPrOpen = prs[0];
+            // override the title to contains the latest aca dep number
+            await this.repoDestination.updatePullRequest(upstreamPrOpen.number, { title, body: `sha:${commit}` });
+            return upstreamPrOpen.number;
+        }
+
+    }
+
+    async fetchContributors(shaFrom, shaTo)  {
+        const mapAuthors = new Map();
+        let upstreamShaFound = true;
+        const listCommits = await this.repoOrigin.listCommits(({sha: shaFrom}))
+        let index = 0;
+        while(upstreamShaFound) {
+            if (listCommits.data[index].sha === shaTo ) {
+                upstreamShaFound = false;
+            } else {
+                mapAuthors.set(listCommits.data[index].author.login, listCommits.data[index].commit.author.name);
+            }
+            index++;
+        }
+        return mapAuthors;
+    }
+
+    async createComment(issueOrPrNumber, head, base, shaOriginHead )  {
+        const shaTo = await this.getShaTo(head, base);
+        const contributors = await this.fetchContributors(shaOriginHead, shaTo);
+        const attemptCount = await this.getCommentAmount(issueOrPrNumber);
+        const commentMsg = this.createCommentBody(contributors, attemptCount);
+        await this.issue.createIssueComment(issueOrPrNumber, commentMsg);
+    }
+
+    createCommentBody(contributors, attemptCount)  {
+        const flattenedContributors = this.flattenContributors(contributors);
+        const attemptMsg = attemptCount <= 5 ? ATTEMPT_MSG[attemptCount] : GIVE_UP_MSG
+        const tmpl = ` Attempt: ${attemptCount+1}
+        you are part of the contributors:
+        ${flattenedContributors}
+        ${attemptMsg}
+        `;
+        return tmpl;
+    }
+
+    flattenContributors(contributors)  {
+        let names = [];
+        for (let key of contributors.keys()) {
+            names.push(`@${key}`)
+        }
+        return names.join(', ');
+    }
+
+    async getShaTo(head, base)  {
+        const { data: closedUpstreamPRs }  = await this.repoDestination.listPullRequests({ state: 'closed', head: `${ORGANISATION}:${head}`, base });
+        const latestClosedUpstream = closedUpstreamPRs[0];
+        const shaTo = latestClosedUpstream.body.split(':')[1].trim();
```

Comment:
do we need to check if there is any `latestClosedUpstream`? first time when it is going to be run, there wont be any I think.

Reply:
Right, added check for this case